### ISSUE TITLE
Refactor: replace `requests` with `urllib3`

### ISSUE THREAD

**Issue by woodruffw** [sigstore/sigstore-python#1040]
`urllib3` has everything we need from `requests` (and is the library beneath it), including connection pooling. It's much more actively maintained and has a more consistent API, so we should switch to it.

**Comment by woodruffw** [comment]
> * urllib3 defaults to system certificates while requests defaults to certifi

Yep, this is a "plus" IMO: the fact that PyPI is a CA bundle delivery mechanism (for `certifi`) has always been one of those "don't look but this is very scary" realities about HTTPS in Python 🙂

**Comment by jku** [comment]
I think that sounds very reasonable.

@sethmlarson if you have the time, would you mind commenting on this one? 
* are we overlooking any potential problems when switching from certifi to system certs
* should we (as library) mention truststore in docs?
* should we (as CLI app) use truststore explicitly?

**Comment by jku** [comment]
The PR linked above makes this urllib3 change for our tuf traffic.

WRT proxy env variables: I added the component (that I kind of think should exist in urllib3) in python-tuf: https://github.com/theupdateframework/python-tuf/blob/develop/tuf/ngclient/_internal/proxy.py. It's pretty small so it's not out of the question to just copy paste here if we do want to support these env vars.
